lib/application.js

```diff
@@ -301,18 +301,16 @@ app.enableAuth = function() {
     var Model = method.ctor;
     var modelInstance = ctx.instance;
 
-    var modelId = modelInstance && modelInstance.id ||
-      // replacement for deprecated req.param()
-      (req.params && req.params.id !== undefined ? req.params.id :
-       req.body && req.body.id !== undefined ? req.body.id :
-       req.query && req.query.id !== undefined ? req.query.id :
-       undefined);
+    var modelId = modelInstance && modelInstance.id || ctx.args.id;
```

Comment:
Hmmm... thats right! I can add in the sharedCtor args to the remote context but I'm not sure that is useful. But I'm not happy with the current mess of param checking (which is http specific for no reason).